### PR TITLE
feat(ai-cookbooks): Add model-based routing cookbook

### PR DESCRIPTION
**Pull request by tomek-labuk**
## Summary

Add cookbook guide for semantic load balancing and intelligent routing using AI Gateway Models.

https://deploy-preview-5515--kongdeveloper.netlify.app/cookbooks/model-based-routing/

### CODE REVIEW

File: app/_cookbooks/model-based-routing.md [hunk 1]
```diff
@@ -0,0 +1,747 @@
...
+{:.no-copy-code}
+
+- **`hide_credentials: true`:** Strips the `apikey` header before forwarding requests to the LLM provider, so API keys never leave {{site.base_gateway}}.
+- **`key_names`:** Defines which header carries the key. The demo uses `apikey` via the OpenAI SDK's `default_headers` parameter.
+
+The same Key Auth configuration applies to both the `model-selection` and `default-llm` Routes. The recipe defines one Consumer (`demo-consumer`) with key `demo-consumer-key` that authenticates to both.
+
+When the DataKit plugin on the default-llm Route calls the model-selection Route internally, it extracts the `apikey` header from the incoming client request and forwards it, so the internal call also passes authentication.
+
+For production deployments, use [{{site.base_gateway}} Vaults](/gateway/secrets-management/) to store API keys:
```
Comment (lmilan) [inline app/_cookbooks/model-based-routing.md:200]:
```suggestion
```

I would say this isn't necessary since it's repeated in the note

File: app/_cookbooks/model-based-routing.md [hunk 2]
```diff
@@ -0,0 +1,747 @@
...
+```bash
+export DECK_OPENAI_SELECTOR_SLM='o3-mini'                # Model selection (SLM)
+export DECK_OPENAI_FAST_MODEL='gpt-4o-mini'              # Fast tier
+export DECK_BEDROCK_SMART_MODEL='anthropic.claude-3-5-sonnet-20241022-v2:0'  # Smart tier (Claude on Bedrock)
+
+# Also export your provider credentials
+export DECK_OPENAI_TOKEN='Bearer sk-...'
+export DECK_AWS_ACCESS_KEY_ID='your-access-key-id'
+export DECK_AWS_SECRET_ACCESS_KEY='your-secret-access-key'
+export DECK_AWS_REGION='us-east-1'
```
Comment (lmilan) [inline app/_cookbooks/model-based-routing.md:411]: Do we need these since they're already in the prereqs?
```suggestion
```

File: app/_cookbooks/model-based-routing.md [hunk 3]
```diff
@@ -0,0 +1,747 @@
...
+Create a file named `multi-provider.yaml` with the configuration below, then sync it to your Control Plane using decK:
+
+```bash
+deck gateway sync recipes/model-based-routing/kong-config/deck/multi-provider.yaml \
+  --konnect-token "${KONNECT_TOKEN}" \
+  --konnect-control-plane-name "${KONNECT_CONTROL_PLANE_NAME}"
+```
+{: data-test-step="block" .collapsible }
+
+The `multi-provider.yaml` file contents:
```
Comment (lmilan) [inline app/_cookbooks/model-based-routing.md]: Should this step come before the deck gateway sync command?

File: app/_cookbooks/model-based-routing.md [hunk 4]
```diff
@@ -0,0 +1,747 @@
...
+                header_value: ${{ env "DECK_OPENAI_TOKEN" }}
+              logging:
+                log_statistics: true
+                log_payloads: false
+              model:
+                provider: openai
+                name: ${{ env "DECK_OPENAI_SELECTOR_SLM" }}
+```
+{% endraw -%}
+{:.no-copy-code}
```
Comment (lmilan) [inline app/_cookbooks/model-based-routing.md]: Why shouldn't we copy this? And should it be collapsible?

File: app/_cookbooks/model-based-routing.md [hunk 5]
```diff
@@ -0,0 +1,747 @@
...
+      "role": "user",
+      "content": "Write a Python function to implement binary search with detailed comments explaining the algorithm, including time complexity analysis and edge case handling."
+    }
+  ],
+  "max_tokens": 500
+}
+EOF
+```
+
+### Test simple prompt routing
```
Comment (lmilan) [inline app/_cookbooks/model-based-routing.md:666]: I'm getting a 500 error for both requests with this message:

> {"request_id":"51c98c589233a86fe10f115f97ebd274","error":"non-2XX response code: 401","message":"node execution error","node":{"index":3,"name":"CALL_MODEL_SELECTOR","type":"call"}}

File: app/_cookbooks/model-based-routing.md (hunk 4 above)
Comment (tomek-labuk) [reply app/_cookbooks/model-based-routing.md]: I think that this is Claude skill's mistake, we'll fix it

**Review by lmilan — APPROVED** [review]
